evm/generators.go

```diff
@@ -1066,6 +1072,229 @@ func Create{{.StructName}}Command() *cobra.Command {
 
 	return cmd
 }
+
+// OperationType represents the type of operation for a Safe transaction
+type OperationType uint8
+
+const (
+	Call         OperationType = 0
+	DelegateCall OperationType = 1
+)
+
+// String returns the string representation of the OperationType
+func (o OperationType) String() string {
+	switch o {
+	case Call:
+		return "Call"
+	case DelegateCall:
+		return "DelegateCall"
+	default:
+		return "Unknown"
+	}
+}
+
+// SafeTransactionData represents the data for a Safe transaction
+type SafeTransactionData struct {
+	To             string        ` + "`" + `json:"to"` + "`" + `
+	Value          string        ` + "`" + `json:"value"` + "`" + `
+	Data           string        ` + "`" + `json:"data"` + "`" + `
+	Operation      OperationType ` + "`" + `json:"operation"` + "`" + `
+	SafeTxGas      uint64        ` + "`" + `json:"safeTxGas"` + "`" + `
+	BaseGas        uint64        ` + "`" + `json:"baseGas"` + "`" + `
+	GasPrice       string        ` + "`" + `json:"gasPrice"` + "`" + `
+	GasToken       string        ` + "`" + `json:"gasToken"` + "`" + `
+	RefundReceiver string        ` + "`" + `json:"refundReceiver"` + "`" + `
+	Nonce          uint64        ` + "`" + `json:"nonce"` + "`" + `
+	SafeTxHash     string        ` + "`" + `json:"safeTxHash"` + "`" + `
+	Sender         string        ` + "`" + `json:"sender"` + "`" + `
+	Signature      string        ` + "`" + `json:"signature"` + "`" + `
+	Origin         string        ` + "`" + `json:"origin"` + "`" + `
+}
+
+const (
+	NativeTokenAddress = "0x0000000000000000000000000000000000000000"
+)
+
+
+func DeployWithSafe(client *ethclient.Client, key *keystore.Key, safeAddress common.Address, factoryAddress common.Address, value *big.Int, txServiceBaseUrl string, deployBytecode []byte, safeOperation OperationType) error {
+	// Generate salt
+	salt, err := GenerateProperSalt(safeAddress)
+	if err != nil {
+		return fmt.Errorf("failed to generate salt: %v", err)
+	}
+
+	abi, err := CreateCall.CreateCallMetaData.GetAbi()
+	if err != nil {
+		return fmt.Errorf("failed to get ABI: %v", err)
+	}
+
+	safeCreateCallTxData, err := abi.Pack("performCreate2", value, deployBytecode, salt)
+	if err != nil {
+		return fmt.Errorf("failed to pack performCreate2 transaction: %v", err)
+	}
+
+	return CreateSafeProposal(client, key, safeAddress, factoryAddress, safeCreateCallTxData, value, txServiceBaseUrl, OperationType(safeOperation))
+}
+
+func GenerateProperSalt(from common.Address) ([32]byte, error) {
```

Comment:
I don't think we need this anymore, since we are not using the `ImmutableCreate2Factory` anymore?

Reply:
We still need to pass a salt to call `safeCreate2` on the CreateCall contract. But you are right; to simplify this function, we no longer need to append the caller's address at the beginning of the salt. 

Would you like to update it, or can we keep it with `ImmutableCreate2Factory` compatibility if we want to use it in the future?

Reply:
I'd rather just take a `--salt` on the `deploy --safe` command.

Take `salt` as an argument from the caller, basically.

I don't like to force a particular salt generation strategy on the user.

Reply:
In this case, do you think it is a problem to keep it that way? 

Because in the actual implementation, if you use `--salt`, it will override this one. 

The autogeneration only happens if the users do not specify a `--salt`

Reply:
@karacurt : I prefer to remove it - the requirement that the salt be prefixed with the caller address is an optional constraint implemented by that particular CREATE2 factory. Other CREATE2 factories will have other means to prevent spoofs, and some may not even put any protections in place. Using that as a default salt just feelds weird to me - and it actually makes it harder to use Safe with `ImmutableCreate2Factory`!

I would have one of two expectations for `--salt`. Either:
1. Use a random salt, or
2. Make `--salt` a required argument

I prefer to make it required.

Reply:
oh great!! i was just afraid to remove the random salt

Reply:
So, do we need to make a CLI to generate a random salt? I think I didn't get this part. 

I am ok with removing the salt standard, but I think it would be useful to have a autogenerated salt if `--salt` not passed

Reply:
Updated!